helix-view/src/handlers/dap.rs

```diff
@@ -274,6 +275,62 @@ impl Editor {
                         self.set_status("Debugged application started");
                     }; // TODO: do we need to handle error?
                 }
+                Event::Terminated(terminated) => {
+                    let restart_args = if let Some(terminated) = terminated {
+                        terminated.restart
+                    } else {
+                        None
+                    };
+                    let restart = restart_args.is_some();
+                    let disconnect_args = Some(DisconnectArguments {
+                        restart: Some(restart),
+                        terminate_debuggee: None,
+                        suspend_debuggee: None,
+                    });
+
+                    let starting_request = debugger.starting_request();
+                    let result = debugger.disconnect(disconnect_args).await;
+                    if result.is_ok() {
+                        if restart {
+                            if let StartingRequest::None = starting_request {
+                                self.set_error("No starting request found, to be used in restarting the debugging session.");
+                            } else {
+                                log::info!("Attempting to restart debug session.");
+                                let restart_args = restart_args.unwrap();
+                                let relaunch_resp =
+                                    if let StartingRequest::Launch = starting_request {
+                                        debugger.launch(restart_args).await
+                                    } else {
+                                        debugger.attach(restart_args).await
+                                    };
+
+                                if let Err(err) = relaunch_resp {
+                                    self.set_error(format!(
+                                        "Failed to restart debugging session: {:?}",
+                                        err
+                                    ));
+                                }
+                            }
```

Comment:
restart should be done with the [`Restart Request`](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Restart) :thinking:

Reply:
I was incorrect in my assumption. the request should be used to _*trigger*_ a restart, rather than "create one". In any case, this should be further elaborated, most likely in a different PR, handling restarting as a whole